libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java

```diff
@@ -65,4 +72,38 @@ private void assertHexRing(int res, String h3Address, String[] children) {
             assertEquals(children[i], ring[positions[i - 1]]);
         }
     }
+
+    public void testNoChildrenIntersecting() {
+        String[] h3Addresses = H3.getStringRes0Cells();
+        String h3Address = RandomPicks.randomFrom(random(), h3Addresses);
+        for (int i = 1; i <= H3.MAX_H3_RES; i++) {
+            h3Addresses = H3.h3ToChildren(h3Address);
+            assertIntersectingChildren(h3Address, h3Addresses);
+            h3Address = RandomPicks.randomFrom(random(), h3Addresses);
+        }
+    }
+
+    private void assertIntersectingChildren(String h3Address, String[] children) {
+        String[] intersectingChildren = H3.h3ToNoChildrenIntersecting(h3Address);
+        for (String child : intersectingChildren) {
```

Comment:
Perhaps the variable could be called `notChild`?

Reply:
done